### DISCUSSION TITLE
What are our options if we are required to retain (and be able to retrieve) GitHub audit logs for a year or more?

### DISCUSSION THREAD

**Discussion by JoeNunnelley** [community/community#121020] [Enterprise]
Because of some regulatory requirements, we may be required to retain our GitHub Cloud audit logs for longer than the maximum 6 months period. I'm curious what solutions others have had success with. I've seen a couple of options and I'm curious what others have done before I commit to one approach or another. The options I'm currently considering are:

1.) log shipping via api to a secured S3 bucket with an 18 month deletion policy
2.) shipping logs to another log aggregator solutions (like Datadog or Splunk)

are their other options I'm not considering?

**Comment by queenofcorgis** [comment]
We generally recommend [Audit Log Streaming](https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/streaming-the-audit-log-for-your-enterprise) for all customers as it gives them a better way to search the data, generate reports, and apply their specific retention policies.
